src/protocol.cpp

```diff
@@ -495,22 +524,74 @@ void CProtocol::CreateAndSendMessage ( const int               iID,
 {
     CVector<uint8_t> vecNewMessage;
     int              iCurCounter;
+    const int        iDataLen = vecData.Size();
 
-    Mutex.lock();
+    // check if message has to be split because it is too large
+    if ( ( iDataLen > MESS_SPLIT_PART_SIZE_BYTES ) && bSplitMessageSupported )
```

Comment:
I'd swap the two sides of the `&&` around as bool truth is an easier computation than comparator truth.  Compiler might do it here as there's no possible side effects.

Reply:
Good idea, I'll change this.